fanout.go

```diff
@@ -124,21 +108,68 @@ func (f *Fanout) ServeDNS(ctx context.Context, w dns.ResponseWriter, m *dns.Msg)
 	return 0, nil
 }
 
+type clientSelector interface {
+	Pick() Client
+}
+
+func (f *Fanout) runWorkers(ctx context.Context, req *request.Request) chan *response {
+	var sel clientSelector
+	if f.serverCount == len(f.clients) {
+		sel = selector.NewSimpleSelector(f.clients)
+	} else {
+		sel = selector.NewWeightedRandSelector(f.clients, f.loadFactor)
+	}
+
+	workerChannel := make(chan Client, f.workerCount)
+	responseCh := make(chan *response, f.serverCount)
+	go func() {
+		defer close(workerChannel)
+		for i := 0; i < f.serverCount; i++ {
+			select {
+			case <-ctx.Done():
+				return
+			case workerChannel <- sel.Pick():
+			}
+		}
```

Comment:
What if the first N servers have failed, but others are available? What if we selected servers that contain the same records m and did not select servers with useful records?

Reply:
> What if the first N servers have failed, but others are available?

It is recommended to use server-count>(clients/2)+1 (quorum) to avoid such problems

> What if we selected servers that contain the same records and did not select servers with useful records?

Sorry, what do you mean here? All servers are expected to have the same records set

Reply:
Now it's fine since we're going to have a block with policy.